### PR TITLE
Support 18n for Braintree error messages

### PR DESCRIPTION
**Pull request by Gamabunta94**
With this PR I want to add the localization for Braintree error messages.
For this purpose I add the Braintree error messages in the localization rails file and show it with the corresponding slug returned from the Braintree response.
I also added Italian translation of the messages.

### CODE REVIEW

**Comment by Gamabunta94** [comment]
It seems the test fails for a timeout error or something else unrelated.

File: app/assets/javascripts/solidus_paypal_braintree/constants.js [hunk 1]
```diff
@@ -6,14 +6,11 @@ SolidusPaypalBraintree = {
       clientTokens: Spree.pathFor('solidus_paypal_braintree/client_token'),
       transactions: Spree.pathFor('solidus_paypal_braintree/transactions')
     },
-
+
```
Comment (tvdeyen) [inline app/assets/javascripts/solidus_paypal_braintree/constants.js]: unnecessary whitespace sneaked into here :)

File: app/assets/javascripts/solidus_paypal_braintree/braintree_errors.js.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+BraintreeError = {
```
Comment (tvdeyen) [inline app/assets/javascripts/solidus_paypal_braintree/braintree_errors.js.erb]: Sorry, just saw that this is a `js.erb` file instead of an html partial. Although not very obviously, but `js.erb` files are very problematic, because changes to the translations won't trigger this file to be re-compiled on next asset compilation. Please make this is html.erb file that gets rendered inside the payment form.

Comment (Gamabunta94) [reply app/assets/javascripts/solidus_paypal_braintree/braintree_errors.js.erb]: Thank you for the feedback.
I changed this behavior and now it should work fine.

**Comment by kennyadsl** [comment]
@tvdeyen can we merge this one? I think it's ok but I don't have permissions to merge here.

**Comment by tvdeyen** [comment]
@kennyadsl merged

**Comment by kennyadsl** [comment]
Thanks!
